### PR TITLE
Add Random, Permute, Linear, Full, and Iota to FusionTranslation

### PR DESCRIPTION
**Pull request by rdspring1**
This PR adds support for random, permute, linear, full, and iota operations. It also includes creating aliased outputs.

### CODE REVIEW

**Comment by rdspring1** [comment]
!build

File: csrc/python_frontend/translation.cpp [hunk 1]
```diff
@@ -679,6 +680,13 @@ class FusionTranslator : public OptInConstDispatch {
       }
     }
 
+    // The min and max reduction operations expect the dtype argument to by
```
Comment (jacobhinkle) [inline csrc/python_frontend/translation.cpp:683]:
```suggestion
    // The min and max reduction operations expect the dtype argument to be
```

File: csrc/python_frontend/translation.cpp [hunk 2]
```diff
@@ -694,7 +702,40 @@ class FusionTranslator : public OptInConstDispatch {
...
+
+  void handlePermute(const LoadStoreOp* lsop) {
+    TensorView* out_tv = lsop->out()->as<TensorView>();
+    const std::vector<IterDomain*>& root_domain = out_tv->domain()->root();
+    std::vector<int64_t> new2old;
+    new2old.reserve(root_domain.size());
+    for (IterDomain* id : out_tv->domain()->logical()) {
+      auto root_id_iter = std::find(root_domain.begin(), root_domain.end(), id);
+      new2old.push_back(std::distance(root_domain.begin(), root_id_iter));
+    }
```
Comment (jacobhinkle) [inline csrc/python_frontend/translation.cpp:730]: Can you use `ir_utils::computePermutation` here?

**Comment by rdspring1** [comment]
!build

File: csrc/python_frontend/translation.cpp (hunk 2 above)
Comment (rdspring1) [reply csrc/python_frontend/translation.cpp]: It works.

**Pull request merged**